components/new_post/user_profile.tsx

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {ReactNode} from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import Constants from 'utils/constants';
+import {fromAutoResponder} from 'utils/post_utils';
+
+import {Post} from '@mattermost/types/posts';
+
+import BotBadge from 'components/widgets/badges/bot_badge';
+import Badge from 'components/widgets/badges/badge';
+import UserProfile from 'components/user_profile';
+import PostHeaderCustomStatus from 'components/post_view/post_header/post_header_custom_status';
+
+type Props = {
+    post: Post;
+    compactDisplay?: boolean;
+    colorizeUsernames?: boolean;
+    isBusy?: boolean;
+    enablePostUsernameOverride?: boolean;
+    isConsecutivePost?: boolean;
+    isBot: boolean;
+    isSystemMessage: boolean;
+    isPostBeingEdited?: boolean;
+    isMobileView: boolean;
+};
+
+const PostUserProfile = (props: Props): JSX.Element | null => {
+    const {post, compactDisplay, isMobileView, isConsecutivePost, isBusy, enablePostUsernameOverride, isBot, isSystemMessage, colorizeUsernames} = props;
+    const isFromAutoResponder = fromAutoResponder(post);
+    const colorize = compactDisplay && colorizeUsernames;
+
+    let userProfile: ReactNode = null;
+    let botIndicator = null;
+
+    const customStatus = (
+        <PostHeaderCustomStatus
+            userId={props.post.user_id}
+            isBot={props.isBot || post.props.from_webhook === 'true'}
+            isSystemMessage={isSystemMessage}
+        />
+    );
+
+    if (compactDisplay || isMobileView) {
+        userProfile = (
+            <UserProfile
+                userId={post.user_id}
+                channelId={post.channel_id}
+                isBusy={isBusy}
+                isRHS={true}
```

Comment:
This still needs to be passed in unfortunately. Even though we aren't rendering the post differently in the RHS, this is still needed for a few things

Reply:
fixing this to not be hardcoded in the e2e PR here
https://github.com/mattermost/mattermost-webapp/pull/11693